proton-j/src/main/java/org/apache/qpid/proton/engine/impl/CollectorImpl.java

```diff
@@ -57,9 +58,13 @@ public void pop()
         }
     }
 
-    public EventImpl put(Event.Type type, Object context)
+    public EventImpl put(EventType type, Object context)
     {
-        if (tail != null && tail.getType() == type &&
+        if (type == null)
+            throw new IllegalArgumentException("Type cannot be null");
+        if (!type.isValid())
+            throw new IllegalArgumentException("Cannot put events of type " + type);
+        if (tail != null && tail.getEventType() == type &&
```

Comment:
I believe it is coaslescing 'duplicate' events.

Your explanation of the changes you did make here reinforces why I think it should probably be changed to equals() now. Before your changes, it wasnt possible for the == check here to do the wrong thing. Following these changes in future it will be possible (even if not that likely) for this to do the wrong thing and it wont necessarily be the easiest thing to find then, whereas it is just now.